debezium-connector-postgres/src/test/java/io/debezium/pipeline/source/snapshot/incremental/PostgresSignalBasedSnapshotChangeEventSourceTest.java

```diff
@@ -0,0 +1,464 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.pipeline.source.snapshot.incremental;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import io.debezium.config.Configuration;
+import io.debezium.connector.SourceInfoStructMaker;
+import io.debezium.connector.postgresql.PostgresConnectorConfig;
+import io.debezium.connector.postgresql.PostgresSignalBasedIncrementalSnapshotChangeEventSource;
+import io.debezium.connector.postgresql.connection.PostgresConnection;
+import io.debezium.doc.FixFor;
+import io.debezium.pipeline.notification.NotificationService;
+import io.debezium.pipeline.source.spi.DataChangeEventListener;
+import io.debezium.pipeline.source.spi.SnapshotProgressListener;
+import io.debezium.relational.Column;
+import io.debezium.relational.ColumnFilterMode;
+import io.debezium.relational.RelationalDatabaseConnectorConfig;
+import io.debezium.relational.Table;
+import io.debezium.relational.TableId;
+
+@RunWith(MockitoJUnitRunner.class)
+public class PostgresSignalBasedSnapshotChangeEventSourceTest {
+
+    public static final String CONNECTION_TEST = "Debezium Test";
+
+    @Mock
+    private NotificationService notificationService;
+
+    protected RelationalDatabaseConnectorConfig config() {
+        return buildConfig(Configuration.create()
+                .with(RelationalDatabaseConnectorConfig.SIGNAL_DATA_COLLECTION, "debezium.signal")
+                .with(RelationalDatabaseConnectorConfig.TOPIC_PREFIX, "core").build());
+    }
+
+    protected RelationalDatabaseConnectorConfig configWithUnionBasedQuery() {
+        return buildConfig(Configuration.create()
+                .with(RelationalDatabaseConnectorConfig.SIGNAL_DATA_COLLECTION, "debezium.signal")
+                .with(RelationalDatabaseConnectorConfig.TOPIC_PREFIX, "core")
+                .with(PostgresConnectorConfig.INCREMENTAL_SNAPSHOT_CHUNK_QUERY_BUILDER_TYPE, "union_based")
+                .build());
+    }
+
+    protected RelationalDatabaseConnectorConfig buildConfig(Configuration configuration) {
+        return new RelationalDatabaseConnectorConfig(configuration, null, null,
+                0, ColumnFilterMode.CATALOG, true) {
```

Comment:
Could you elaborate please why do we need this parameters and why we can not just use a constructor of `PostgresConnectorConfig`? Also I do not think that `ColumnFilterMode.CATALOG` is anywhere for Postgres, do I miss something?

Reply:
It was changed to use `PostgresConnectorConfig` now

Reply:
And why do we need this method now? Can we just use `PostgresConnectorConfig`? I have not looked into the details but at least tests are passing like that and the overwritten methods look like just stubs.

Reply:
Yes, tests work without overridden methods. Removing the `buildConfig` method.